src/main/java/frc/robot/devices/ColorSensor.java

```diff
@@ -0,0 +1,108 @@
+package frc.robot.devices;
+
+import com.revrobotics.ColorMatch;
+import com.revrobotics.ColorMatchResult;
+import com.revrobotics.ColorSensorV3;
+import edu.wpi.first.util.sendable.Sendable;
+import edu.wpi.first.util.sendable.SendableBuilder;
+import edu.wpi.first.wpilibj.I2C;
+import edu.wpi.first.wpilibj.util.Color;
+
+/**
+ * Contains methods for interfacing with the REV Color Sensor V3.
+ */
+public class ColorSensor implements Sendable {
+
+    private ColorSensorV3 colorSensor;
+    private ColorMatch colorMatcher;
+
+    private final Color blueTarget;
+    private final Color redTarget;
+
+    /**
+     * Creates a new ColorSensor object.
+     */
+    public ColorSensor() {
+        colorSensor = new ColorSensorV3(I2C.Port.kOnboard);
+        colorMatcher = new ColorMatch();
+
+        blueTarget = new Color(0.13989, 0.3818, 0.4787);
+        redTarget = new Color(0.5510, 0.33178, 0.1176);
+
+        colorMatcher.addColorMatch(blueTarget);
+        colorMatcher.addColorMatch(redTarget);
+    }
+
+    /**
+     * Gets the red value detected by the sensor.
+     *
+     * @return red the red value detected by the sensor
+     */
+    public double getRed() {
+        return colorSensor.getColor().red;
+    }
+
+    /**
+     * Gets the green value detected by the sensor.
+     *
+     * @return green the green value detected by the sensor
+     */
+    public double getGreen() {
+        return colorSensor.getColor().green;
+    }
+
+    /**
+     * Gets the blue value detected by the sensor.
+     *
+     * @return blue the blue value detected by the sensor
+     */
+    public double getBlue() {
+        return colorSensor.getColor().blue;
+    }
+
```

Comment:
add a function to return the proximity reading (also possibly make color sensor extend lidar for convenience and interoperability)

Reply:
I implemented this on `main`.